vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/AbstractExpressionProcessors.java

```diff
@@ -0,0 +1,19 @@
+package org.vividus.bdd.expression;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public abstract class AbstractExpressionProcessors implements IExpressionProcessor
+{
+    @Override
+    public Optional<String> execute(String expression)
+    {
+        return getExpressionProcessors().stream()
+                .map(processor -> processor.execute(expression))
+                .filter(Optional::isPresent)
+                .findFirst()
+                .orElseGet(Optional::empty);
+    }
+
+    protected abstract Collection<IExpressionProcessor> getExpressionProcessors();
```

Comment:
```java
public class DelegatingExpressionProcessor implements IExpressionProcessor
{
    private final Collection<IExpressionProcessor> delegates;

    public DelegatingExpressionProcessor(Collection<StoryReporter> delegates)
    {
        this.delegates = delegates;
    }

    @Override
    public Optional<String> execute(String expression)
    {
        return delegates.stream()
                .map(processor -> processor.execute(expression))
                .filter(Optional::isPresent)
                .findFirst()
                .orElseGet(Optional::empty);
    }
}
```
?

Reply:
There is not way to refer to instance methods and fields from super constructor invocation, I could make those methods static, but then I can't refer to the beans I injected to do datetime operations